contracts/libraries/Pool.sol

```diff
@@ -138,6 +145,12 @@ library Pool {
         self.slot0.hookFees = hookFees;
     }
 
+    /// @notice Only dynamic fee pools may update the swap fee.
+    function _setSwapFee(State storage self, uint24 swapFee) internal {
+        if (self.slot0.sqrtPriceX96 == 0) revert PoolNotInitialized();
```

Comment:
maybe require that `self.key.isDynamicFee()`?

Reply:
Re this comment [here](https://github.com/Uniswap/v4-core/pull/360#discussion_r1378177399), the change I made actually moves the check up a level (so its directly inside updateDynamicSwapFee). I did this because we have a duplicate check in initialize if the check is actually inside _fetchDynamicSwapFee. 

With this change, then by the time we call `setSwapFee` we have already done that check.

So I think just this change fixes both this comment and the one below. But lmk what you think.